docs/howto/graph-api/intro.md

```diff
@@ -0,0 +1,151 @@
+# The Graph(-style) API
+
+A [GraphQL](https://graphql.org/)-like API is implemented to allow flexible and accurate data fetching.
+
+??? note "technical details"
+    It is categorized as a `GraphQL`-**like** API implemented within the `REST` style framework `FastAPI`.
+    Because `GraphQL` requires static, explicitly defined schemas ahead of time while `NOMAD` supports data with dynamic schema,
+    it cannot be implemented directly using existing `GraphQL` tools.
+    As a result, there are no GUI tools available unfortunately.
+
+## Overview
+
+In general, `REST` is good for simple data fetching, while the project gets more complex, APIs in REST style become more complex and less flexible.
+When building a complex page, often a single request is not enough, and multiple requests are needed to fetch all the necessary data.
+`GraphQL` aims to solve this over-/under-fetching problem so that both performance and bandwidth can be optimized.
+
+You walk into a restaurant, go through the menu, and order several dishes at once.
+The kitchen prepares all the dishes and serves them to you, nothing more, nothing less.
+This is effectively how `GraphQL` works.
+
+In `NOMAD`, we mimic this behaviour with a `GraphQL`-like API.
+The only endpoint involved is `/graph/query`.
+All the magic happens there.
+But before that, we shall first explain some basic concepts.
+
+## Basic Data Fetching
+
+Imagine there is an example upload with the upload ID `example_upload_id`.
+The metadata of this upload is stored in `MongoDB`.
+In our mental model, it is possible to organize all uploads in a tree structure, under the root `uploads`.
+This better helps us to define interactions among different data structures (to be explained later).
+
+If one uses the endpoint `/uploads/{upload_id}` to fetch the upload metadata,
+the response would look like the following (after some restructuring according to the mental model).
+
+```json title="an example upload"
```

Comment:
We don't use titles in our code snippets, remove these from each snippet on this page.

Reply:
It helps clearing potential confusions, isn't it?

Reply:
Not used anywhere else, so don't add here.